Mage.Sets/src/mage/cards/r/RalLeylineProdigy.java

```diff
@@ -0,0 +1,234 @@
+package mage.cards.r;
+
+import mage.MageIdentifier;
+import mage.MageObject;
+import mage.MageObjectReference;
+import mage.ObjectColor;
+import mage.abilities.Ability;
+import mage.abilities.LoyaltyAbility;
+import mage.abilities.common.EntersBattlefieldAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.condition.common.PermanentsOnTheBattlefieldCondition;
+import mage.abilities.decorator.ConditionalOneShotEffect;
+import mage.abilities.effects.AsThoughEffectImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DamageMultiEffect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.cost.SpellsCostReductionAllEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.counters.CounterType;
+import mage.filter.FilterCard;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterInstantOrSorceryCard;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+import mage.filter.predicate.mageobject.ColorPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.common.TargetAnyTargetAmount;
+import mage.util.CardUtil;
+import mage.watchers.common.SpellsCastWatcher;
+
+import java.util.*;
+
+/**
+ * @author Susucr
+ */
+public final class RalLeylineProdigy extends CardImpl {
+
+    private static final FilterPermanent filterBlue = new FilterPermanent("blue permanent other than {this}");
+    private static final Condition condition = new PermanentsOnTheBattlefieldCondition(filterBlue, true);
+
+    static {
+        filterBlue.add(new ColorPredicate(ObjectColor.BLUE));
+        filterBlue.add(AnotherPredicate.instance);
+    }
+
+    public RalLeylineProdigy(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.PLANESWALKER}, "");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.RAL);
+        this.setStartingLoyalty(2);
+
+        this.color.setBlue(true);
+        this.color.setRed(true);
+        this.nightCard = true;
+
+        // Ral, Leyline Prodigy enters the battlefield with an additional loyalty counter on him for each instant and sorcery spell you've cast this turn.
+        this.addAbility(new EntersBattlefieldAbility(new RalLeylineProdigyCounterEffect()));
```

Comment:
Why not a standard effect here? Usage of custom effects/abilities should be kept to a minimum.

![IMG_0219](https://github.com/magefree/mage/assets/8344157/072c5333-236b-4600-ae44-b30d7f6e11e7)

Reply:
Well I need a custom DynamicValue then.

Reply:
Thats correct usage — standard effects/abilities with customizable options like filter, target, condition.

If you can’t find compatible effect then extract it from same card to standard class. If it do not exists at all then create custom in card class.

Reply:
And next hint: if you use dynamic value then card hint must be added in most use cases :)